app/src/main/java/ip/team13/petowner/data/repository/AuthRepository.kt

```diff
@@ -11,26 +11,37 @@ class AuthRepository(
     private val preferences: Preferences
 ) {
 
-    suspend fun register(email: String, password: String, inviteCode: String) {
+    suspend fun register(
+        email: String, password: String, inviteCode: String? = null, onSuccess: (() -> Unit)? = null
+    ) {
 
         try {
             val response = petOwnerAPI.register(
                 RegisterForm(email, password, preferences.getFCMToken() ?: "", inviteCode)
             )
-            preferences.saveBearerToken(response.userToken)
-            preferences.saveUserId(response.userId)
+            response.userToken?.let { preferences.saveBearerToken(it) }
+            response.userId?.let { preferences.saveUserId(it) }
+
+            if (!response.userToken.isNullOrEmpty() && response.userId != null) {
+                onSuccess?.invoke()
+            }
+
         } catch (e: Exception) {
             e.message?.logError()
         }
     }
 
-    suspend fun login(email: String, password: String) {
+    suspend fun login(email: String, password: String, onSuccess: (() -> Unit)? = null) {
         try {
             val response = petOwnerAPI.login(
                 LoginForm(email, password, preferences.getFCMToken() ?: "")
             )
-            preferences.saveBearerToken(response.userToken)
-            preferences.saveUserId(response.userId)
+            response.userToken?.let { preferences.saveBearerToken(it) }
+            response.userId?.let { preferences.saveUserId(it) }
+
+            if (!response.userToken.isNullOrEmpty() && response.userId != null) {
```

Comment:
Why do we need business logic inside the Data layer? Also, there is no need for callbacks, we can just add the "success" code after the `login` call in VM